From: SeongJae Park <sj@kernel.org>
Date: Fri, 28 Nov 2025 15:41:38 -0800
Subject: [PATCH] mm/damon: remove damon_report_filter->write

The field is unnecessary, since the type itself is indicating it is for
write.  Remove it.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 6 +-----
 mm/damon/core.c       | 1 -
 2 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 5bc7ed94f0cf..a62f3ba980df 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -862,17 +862,13 @@ enum damon_report_filter_type {
  * @matching:	Whether is is for condition-matching reports.
  * @allow:	Whether to include or excludie the @matching reports.
  * @cpus:	Access-originated CPUs if @type is DAMON_FILTER_TYPE_CPUS.
- * @write:	If the access was writing if @type is DAMOS_FILTER_TYPE_WRITE.
  * @list:	List head for siblings.
  */
 struct damon_report_filter {
 	enum damon_report_filter_type type;
 	bool matching;
 	bool allow;
-	union {
-		cpumask_t cpus;
-		bool write;
-	};
+	cpumask_t cpus;
 	struct list_head list;
 };
 
diff --git a/mm/damon/core.c b/mm/damon/core.c
index 80a5edf03f93..bd4bdc9bd766 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1338,7 +1338,6 @@ static void damon_commit_report_filter_arg(struct damon_report_filter *dst,
 		dst->cpus = src->cpus;
 		break;
 	case DAMON_FILTER_TYPE_WRITE:
-		dst->write = src->write;
 		break;
 	default:
 		break;
-- 
2.47.3

